date/index.js

```diff
@@ -306,6 +306,29 @@ export function dateI18n( dateFormat, dateValue = new Date(), gmt = false ) {
 	return format( dateFormat, dateMoment );
 }
 
+/**
+ * Human time difference (like `human_time_diff()` in PHP).
+ *
+ * @param  {?(Date|String|Number|Moment|null)} from    Value parsable by moment.js or 'now'. Default is now.
+ * @param  {?(Date|String|Number|Moment|null)} to      Value parsable by moment.js or 'now'. Default is now.
+ * @param  {?Boolean}                          verbose True to add an an 'ago' (past) or 'from now' (future) suffix.
+ *                                                     Defaults to false so it matches `human_time_diff()` in PHP,
+ *                                                     which doesn't support this additional argument yet.
+ *
+ * @return {String}                                    Difference; e.g., 3 days, in 3 days, 3 days ago, 3 days from now.
+ */
+export function humanTimeDiff( from, to, verbose = false ) {
+	// Establish 'from' and 'to' times.
+	from = ! from || from === 'now' ? new Date() : from;
+	to = ! to || to === 'now' ? new Date() : to;
+	// Convert to moment.
+	const fromMoment = moment( from );
+	// Set the locale.
+	fromMoment.locale( window._wpDateSettings.l10n.locale );
```

Comment:
Something that worries me here, is that I'm not sure these are translated properly because we do not use the built-in moment translations but we use generated translations from WP i18n files.

Reply:
If it helps, the relevant translation strings for this Moment functionality can be [found here in Gutenberg](https://github.com/WordPress/gutenberg/blob/8e871822d303a1ca8749fa9257a0799a8e874ec8/date/index.js#L203-L204), which comes from [this localization](https://github.com/WordPress/gutenberg/blob/8e871822d303a1ca8749fa9257a0799a8e874ec8/lib/client-assets.php#L113-L116).

To your point, I think the remaining issue is that [these strings](https://github.com/WordPress/gutenberg/blob/8e871822d303a1ca8749fa9257a0799a8e874ec8/date/index.js#L205-L215) have not yet been translated in Gutenberg. I'll see if I can resolve that.

Reply:
I'm not sure I follow - doesn't this line set the moment.locale and thus rely on moment.js translation functionality for the difference string? In my case for example `_wpDateSettings.l10n.locale
` is `en_US`. https://momentjs.com/docs/#/i18n/

Reply:
Thanks for catching that! Yep, that line can be removed. Looks to me like that same line can be removed from the existing `dateI18n()` utility also. The locale is set at the end of this file already.

Reply:
Actually, no. Those lines can stay and are correct. The `setupLocale()` method is defining the translations for `_wpDateSettings.l10n.locale`, and then _restoring_ the original default Moment locale. So it's still necessary to set the locale again before using Moment elsewhere in this file.

The translation strings are already prepared for this, given the `setupLocale()` function that runs in this file. So the following:

```js
fromMoment.locale( window._wpDateSettings.l10n.locale )
```

Is setting the locale, which is preconfigured ahead of time by `setupLocale()` in this file.

Reply:
I'm going to translate those remaining strings for relative date calculations in just a sec.

Reply:
To understand this better, see: https://momentjs.com/docs/#/customization/
And then look at what `setupLocale()` does in [this Gutenberg file](https://github.com/jaswrks/gutenberg/blob/3236f1ac92b3defc75c6fc49f5f6632cfe14749b/date/index.js#L176), and [where `setupLocale()`](https://github.com/jaswrks/gutenberg/blob/3236f1ac92b3defc75c6fc49f5f6632cfe14749b/date/index.js#L344) is called from when the file is imported.